app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -841,20 +847,28 @@ public function setResponseContextHeaders()
             ->setHeader('X-Bolt-Plugin-Version', $context_info["Bolt-Plugin-Version"], true);
     }
 
-    public function getQuantityCheck()
+
+    /**
+     * Determines whether the cart has either all items available in the requested quantities,
+     * or, if not, those items are eligible for back order.
+     *
+     * @return bool true if the store can accept an order for all items in the cart,
+     *              otherwise, false
+     */
+    public function storeHasAllCartItems()
     {
+        /* @var Mage_Sales_Model_Quote $cart_quote */
+        $cart_quote = Mage::helper('checkout/cart')->getCart()->getQuote();
 
-        $quoteCart = Mage::helper('checkout/cart')->getCart()->getQuote();
-        $QtyFlagCheck = false;
+        foreach ($cart_quote->getAllItems() as $cart_item) {
+            $_product = Mage::getModel('catalog/product')->load($cart_item->getProductId());
+            $stock_info = Mage::getModel('cataloginventory/stock_item')->loadByProduct($_product);
```

Comment:
Can you clarify why we should load the product data through "load"?
I thought that we have the correct data after calling  `$cart_item->getProduct()` and `$cart_item->getProduct()->getStockItem()`.